packages/core/src/controllers/TransactionsController.ts

```diff
@@ -0,0 +1,108 @@
+import { DateUtil } from '@web3modal/common'
+import type { Transaction } from '@web3modal/common'
+import { proxy, subscribe as sub } from 'valtio/vanilla'
+import { BlockchainApiController } from './BlockchainApiController.js'
+import { OptionsController } from './OptionsController.js'
+import { AccountController } from './AccountController.js'
+import { EventsController } from './EventsController.js'
+import { SnackController } from './SnackController.js'
+
+// -- Types --------------------------------------------- //
+type TransactionByYearMap = Record<number, Transaction[]>
+
+export interface TransactionsControllerState {
+  transactions: Transaction[]
+  transactionsByYear: TransactionByYearMap
+  loading: boolean
+  empty: boolean
+  next?: string
+}
+
+// -- State --------------------------------------------- //
+const state = proxy<TransactionsControllerState>({
+  transactions: [],
+  transactionsByYear: {},
+  loading: false,
+  empty: false,
+  next: undefined
+})
+
+// -- Controller ---------------------------------------- //
+export const TransactionsController = {
+  state,
+
+  subscribe(callback: (newState: TransactionsControllerState) => void) {
+    return sub(state, () => callback(state))
+  },
+
+  async fetchTransactions() {
+    const projectId = OptionsController.state.projectId
+    const accountAddress = AccountController.state.address
+
+    if (!projectId || !accountAddress) {
+      throw new Error("Transactions can't be fetched without a projectId and an accountAddress")
+    }
+
+    state.loading = true
+
+    try {
+      await BlockchainApiController.fetchTransactions({
+        account: accountAddress,
+        projectId,
+        cursor: state.next
+      }).then(response => {
```

Comment:
Should this be not `.then()` but rather `const response = await BlockchainApiController...` ?

Reply:
That works as well. The `then` will be working only if the fetch doesn't throw any errors. So the we could reach the same functionality with `const response = await BlockchainApiController ` approach as well. It's just different style of handling async functions.

Seems like we don't use `then` approach so I'll update it now